src/sources/journald.rs

```diff
@@ -124,21 +137,54 @@ fn create_event(record: Record) -> Event {
     log.into()
 }
 
+trait JournalCursor {
```

Comment:
looks like below you bound this with `Iterator<Item = Result<Record, Error>>` would it make sense to bound this trait by that so that they are coupled?

Reply:
I'm not sure it adds much, and neither trait is used by the other directly so they aren't strictly tied. If you think it makes sense I'll defer to your judgment.